packages/account-v2/src/modules/FinancialAssessment/FinancialAssessmentForm.tsx

```diff
@@ -0,0 +1,174 @@
+import IcSuccess from '../../assets/status-message/ic-success.svg';
+import { DemoMessage } from '../../components/DemoMessage';
+import { ACCOUNT_V2_DEFAULT_ROUTE, ACCOUNT_V2_ROUTES, DERIV_GO_URL, P2P_URL } from '../../constants/routes';
+import { isNavigationFromDerivGO, isNavigationFromP2P } from '../../utils/platform';
+import { FinancialAssessmentFields } from '../FinancialAssessmentFields';
+
+export const FinancialAssessmentForm = () => {
+    const { data: activeAccount } = useActiveTradingAccount();
+    const { data: accountStatus } = useAccountStatus();
+    const { isEUCountry } = useIsEuRegion();
+    const { isMobile } = useDevice();
+    const history = useHistory();
+
+    Modal.setAppElement(ACCOUNT_MODAL_REF);
+
+    const redirect = isNavigationFromDerivGO() || isNavigationFromP2P();
+    const redirectPlatformName = isNavigationFromDerivGO() ? 'Deriv GO' : 'Deriv P2P';
+    const redirectUrl = isNavigationFromDerivGO() ? DERIV_GO_URL : P2P_URL;
+
+    const {
+        is_authenticated: isAuthenticated,
+        is_financial_information_not_complete: isFinancialInformationNotComplete,
+    } = accountStatus || {};
+
+    const {
+        data: financialAssessment,
+        error: fetchError,
+        isLoading: isFinancialAssessmentLoading,
+        mutation: {
+            error: financialAssessmentUpdateError,
+            isLoading: isFinancialAssessmentUpdating,
+            isSuccess: isFinancialAssessmentUpdateSuccess,
+        },
+        update: updateFinancialAssessment,
+    } = useFinancialAssessment();
+
+    const {
+        account_turnover: accountTurnover,
+        education_level: educationLevel,
+        employment_industry: employmentIndustry,
+        employment_status: employmentStatus,
+        estimated_worth: estimatedWorth,
+        income_source: incomeSource,
+        net_income: netIncome,
+        occupation,
+        source_of_wealth: sourceOfWealth,
+    } = financialAssessment;
+
+    const initialValues = {
+        accountTurnover,
+        educationLevel,
+        employmentIndustry,
+        employmentStatus,
+        estimatedWorth,
+        incomeSource,
+        netIncome,
+        occupation,
+        sourceOfWealth,
+    };
+
+    type TFinancialAssessmentFormValues = typeof initialValues;
+
+    const handleFormSubmit = async (values: TFinancialAssessmentFormValues) => {
+        updateFinancialAssessment({
+            account_turnover: values.accountTurnover,
+            education_level: values.educationLevel,
+            employment_industry: values.employmentIndustry,
+            employment_status: values.employmentStatus,
+            estimated_worth: values.estimatedWorth,
+            income_source: values.incomeSource,
+            net_income: values.netIncome,
+            source_of_wealth: values.sourceOfWealth,
+            ...(shouldHideOccupation(values.employmentStatus) ? {} : { occupation: values.occupation }),
```

Comment:
instead of setting empty object, will this work here?
`...(!shouldHideOccupation(values.employmentStatus) && { occupation: values.occupation }),`

Reply:
Thought it might give an error on destructuring boolean value.
Updated it in this [PR](https://github.com/binary-com/deriv-app/pull/14445)